src/main/java/org/apache/maven/plugins/pmd/exec/PmdExecutor.java

```diff
@@ -323,6 +307,26 @@ private PmdResult run() throws MavenReportException
         return new PmdResult( new File( request.getTargetDirectory(), "pmd.xml" ), request.getOutputEncoding() );
     }
 
+    /**
+     * Gets the errors as a single string. Each error is in its own line.
+     * @param withDetails if <code>true</code> then add the error details additionally (contains e.g. the stacktrace)
+     * @return the errors as string
+     */
+    private String getErrorsAsString( List<Report.ProcessingError> errors, boolean withDetails )
+    {
+        List<String> errorsAsString = new ArrayList<>( errors.size() );
+        for ( Report.ProcessingError error : errors )
+        {
+            errorsAsString.add( error.getFile() + ": " + error.getMsg() );
+            if ( withDetails )
+            {
+                errorsAsString.add( error.getDetail() );
```

Comment:
No space betweeen msessage and detail?

Reply:
That's fine, because we collect separate lines (errorsAsString is a `List<String>`) and later join them.